src/main/java/cms/gov/madie/terminology/controller/VsacController.java

```diff
@@ -59,7 +73,42 @@ protected String serializeFhirValueset(ValueSet fhirValueSet) {
 
   @PutMapping(path = "/validations/codes", produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<List<CqlCode>> validateCodes(
-      @RequestBody List<CqlCode> cqlCodes, @RequestParam String tgt) {
-    return ResponseEntity.ok().body(vsacService.validateCodes(cqlCodes, tgt));
+      Principal principal, @RequestBody List<CqlCode> cqlCodes) {
+    final String username = principal.getName();
+    Optional<UmlsUser> umlsUser = vsacService.findByHarpId(username);
+    if (umlsUser.isPresent() && umlsUser.get().getApiKey() != null) {
+      return ResponseEntity.ok().body(vsacService.validateCodes(cqlCodes, umlsUser.get().getTgt()));
+    }
+    return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
Do you think it's useful to return 401 instead of 404?

Reply:
Getting an API-KEY or TGT or service ticket, all 3 comes under authentication. 404 indicates that "code or code system is not found" but that's not the case.

Reply:
@RohitKandimalla This is to tell that there is no database record for the user. That's why.

Reply:
There's no complimentary error in the value set validation method. Any idea what it returns when the user's creds are missing?

Reply:
@jkotanchik-SB @RohitKandimalla I changed it to unauthorized, if user credentials are missing.